### ISSUE TITLE
Depth image value range

### ISSUE THREAD

**Issue by JRauer** [mmatl/pyrender#10]
Hello,
I tried your example code to render the object in color and depth but can't get the correct result (see screenshot). In the depth image, the background is white (Value 0) and the Object has values between 1.13 and 1.2 - although it should be a float32 image. Do I have to do a normalization or something to get the correct result?
Thank you in advance!

**Comment by mmatl** [comment]
Hi @JRauer,

I can't see your screenshot, sorry! Any place with undefined depth will have depth == 0 (this makes it easy to find invalid depth values). The depths should be real-valued (i.e. a depth of 1.13 means that that pixel is exactly 1.13 meters away from the camera origin).

**Comment by mmatl** [comment]
@JRauer Any update on your issues with depth?

**Comment by JRauer** [comment]
Hi @mmatl ,

thanks for your fast reply! I'm sorry, there went something wrong while uploading the screenshot I guess. Invalid depth = 0 makes sense to me. Also displaying the distance in meters in the depth map makes sense, but it does not seem to be calculated correctly. I tried the example with a camera-pose-translation of x=0.3, z=0.3 (left image) and x=0.5, z=0.5 (right image). If the distance would be in meters, the distance in the right image should be bigger than in the left image - but as you can see at the values in the lower right corners (mouse-cursor at red dot on object) the values don't seem to be correct.
![depth-values](https://user-images.githubusercontent.com/16468350/53792172-57c23600-3f2b-11e9-8683-edb4adf2a465.png)

**Comment by mmatl** [comment]
@JRauer Thanks for the minimal working example! I investigated and found the source of the bug. My code was working if you defined a `zfar` for your camera, but I had a mistake in my implementation of depth for infinite perspective projection cameras. This is fixed in https://github.com/mmatl/pyrender/commit/c85a0348fcb37cacfbc9b27ceba0ef17e9a90ba0 and has been released in version 0.1.14. Sorry for the trouble, let me know if that fixes things for you!
